src/wasm/wasm-validator.cpp

```diff
@@ -44,7 +43,7 @@ inline std::ostream& printModuleComponent(T curr, std::ostream& stream) {
 // Extra overload for Expressions, to print type info too
 inline std::ostream& printModuleComponent(Expression* curr,
                                           std::ostream& stream) {
-  WasmPrinter::printExpression(curr, stream, false, true) << std::endl;
+  stream << *curr << '\n';
```

Comment:
Am I correct that this will previously printed the expression in "full" mode but now it does not? If so, is this intended?

Reply:
Yes, I think this is ok to do. (But it makes this not NFC.)

The user can always set `BINARYEN_PRINT_FULL=1` to get "full" printing (i.e. print types on each line) when necessary. I'm not sure how helpful it is to do by default - in fact it is sometimes annoying.